File: FE/src/services/venueService.js
import axios from "axios";

const BASE_URL = "http://localhost:8081/api/venues";

export const getAllVenues = () => {
  return axios.get(BASE_URL);
};

export const getVenueById = (id) => {
  return axios.get(`${BASE_URL}/${id}`);
};

export const addVenue = (venue) => {
  return axios.post(BASE_URL, venue);
};

export const updateVenue = (id, venue) => {
  return axios.put(`${BASE_URL}/${id}`, venue);
};

export const deleteVenue = (id) => {
  return axios.delete(`${BASE_URL}/${id}`);
};

File: Venue_Service/src/main/java/com/EventManagementSystem/Venue_Service/service/VenueService.java
package com.EventManagementSystem.Venue_Service.service;

import com.EventManagementSystem.Venue_Service.entity.Venue;

import java.util.List;

public interface VenueService {
    void addVenue(Venue venue);
    Venue getVenueById(Long id);
    List<Venue> getAllVenues();
    void updateVenue(Long id, Venue venue);
    void removeVenue(Long id);
}